modules/tide_dashboard/tide_dashboard.module

```diff
@@ -37,6 +37,8 @@ function tide_dashboard_user_login($account) {
   $destination = \Drupal::service('redirect.destination')->get();
   if (!(strpos($destination, '/oauth/authorize') !== FALSE)) {
     $request = \Drupal::service('request_stack')->getCurrentRequest();
-    $request->query->set('destination', '/admin/workbench');
+    if (($request->query->has('destination')) === FALSE) {
+      $request->query->set('destination', '/admin/workbench');
+    }
   }
```

Comment:
Hey @edyuenyw,

Can we simplify the logic in this a bit?

Instead of checking for the existence of the destination /oauth/authorize, then checking the parameter again, we could just do:

```
function tide_dashboard_user_login($account) {
    if (($request->query->has('destination')) === FALSE) {
      $request->query->set('destination', '/admin/workbench');
    }
}
```

This will prevent a redirect to the workbench whenever the original request already has a different destination set.

Reply:
That looks cleaner. Will start making the change.